sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1868,9 +1868,9 @@ class Dataset[T] private[sql](
  /**
   * Define (named) metrics to observe on the Dataset. This method returns an 'observed' Dataset
   * that returns the same result as the input, with the following guarantees:
-  * - It will compute the defined aggregates (metrics) on all the data that is flowing through the
+  *  - It will compute the defined aggregates (metrics) on all the data that is flowing through the
   *   Dataset at that point.
-  * - It will report the value of the defined aggregate columns as soon as we reach a completion
+  *  - It will report the value of the defined aggregate columns as soon as we reach a completion
```

Comment:
To make completely fix it, you gotta use HTML tag to render properly in both javadoc and scaladoc. See https://github.com/apache/spark/blob/9dedfeffaa27a00f719b5a50ce1692f6a0ad665d/sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala#L527-L536 as an example.

You can simply revert this change too here.